### PR TITLE
feat: Add StringLiteral.

### PR DESCRIPTION
**Pull request by lhruby**
# Description
Add `StringLiteral` class to allow using strings as non-type template parameters (NTTP).

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x ] I have performed a self-review of my code.
- [x ] If it is a core feature, I have added thorough tests.
- [ ] If this is a new component I have added examples.
- [ ] I updated the README and related documentation.

### CODE REVIEW

File: modules/utils/tests/string_literal_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+#include "hephaestus/utils/string/string_literal.h"
+
+// NOLINTNEXTLINE(google-build-using-namespace)
+using namespace ::testing;
+
+namespace heph::utils::string::tests {
+
+/// Generate a random string of characters, including special case characters and numbers.
+auto randomString(std::mt19937_64& mt) -> std::string {
```
Comment (filippobrizzi) [inline modules/utils/tests/string_literal_tests.cpp:17]: this feels like a duplicate from the other PR, maybe you can extract it into a common file in the `test` folder

**Review by filippobrizzi — APPROVED** [review]

File: modules/utils/tests/string_literal_tests.cpp (hunk 1 above)
Comment (lhruby) [reply modules/utils/tests/string_literal_tests.cpp:17]: This was a mistake, it's actually not used. Removed.

File: modules/utils/include/hephaestus/utils/string/string_utils.h [hunk 2]
```diff
@@ -19,17 +19,17 @@ namespace heph::utils {
...
-  if (end_pos != std::string_view::npos && include_end_token) {
-    end_pos += end_token.size();
-  }
-
-  return (start_pos != std::string_view::npos) ? str.substr(start_pos, end_pos - start_pos) :
-                                                 str.substr(0, end_pos);
-}
-
-}  // namespace heph::utils
+[[nodiscard]] auto truncate(std::string_view str, std::string_view start_token,
```
[the remark this answers is not in the document]
Comment (lhruby) [reply modules/utils/include/hephaestus/utils/string/string_utils.h]: Yes, updated.

**Comment by filippobrizzi** [comment]
@lhruby if you rebase now it should fix CI

**Pull request merged**